discretisedfield/field.py

```diff
@@ -2021,25 +2020,17 @@ def diff(self, direction, order=1):
                     padded_array, self.mesh.cell[direction_idx], axis=direction_idx
                 )
             else:
-                # The derivative is computed using accuracy of 2 everywhere
-                diff_fd = fd.FinDiff(direction_idx, self.mesh.cell[direction_idx], 1)
-                derivative_array = diff_fd(padded_array)
-
-        elif order == 2:
-            if self.mesh.n[direction_idx] < 4:
-                # The derivative is computed using the central difference
-                # with forward/backward difference at the boundaries.
                 if self.mesh.bc == "":
                     # Pad with specific values so that the same finite difference
                     # stencil can be used across the whole array
                     # central difference = forward difference
-                    # f(1) + f(-1) - 2 f(0) = f(2) + f(0) - 2 f(1)
-                    # f(-1) = f(2) - 3 f(1) + 3f(0)
-
+                    # 0.5 * f(1) - 0.5 * f(-1) = - 0.5 f(2) + 2 f(1) - 1.5 f(0)
```

Comment:
Understanding these coefficients (or where they come from) could be a bit difficult. Mabye a reference to findiff (and mentioning why we cannot use it directly) could be useful?